# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/BackgroundParentImpl.cpp
# Commit: e89ac54f184d
# Full Hash: e89ac54f184df504337b7394428ab6307578eec1
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2023-10-04 03:53:37
# Regressor Bug: 1855138
# File Overlap Count: 3
# Description:
#   Bug 1855138 - Add BackgroundParent::KillHardAsync method; r=nika,ipc-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D189482
# ==============================================================================

diff -r 23e1dfcc8598 -r e89ac54f184d ipc/glue/BackgroundParentImpl.cpp
--- a/ipc/glue/BackgroundParentImpl.cpp	Tue Oct 03 18:25:15 2023 +0000
+++ b/ipc/glue/BackgroundParentImpl.cpp	Tue Oct 03 18:40:30 2023 +0000
@@ -816,6 +816,11 @@
     return IPC_OK();
   }
 
+  // XXX It's now possible to call KillHardAsync on the PBackground thread, so
+  // the checks can be done directly here.
+
+  // XXX Once BackgroundParent::ProcessingError calls KillHardAsync, we can
+  // just return IPC_FAIL() if a check fails.
   RefPtr<CheckPrincipalRunnable> runnable =
       new CheckPrincipalRunnable(parent.forget(), aPrincipalInfo, aOrigin);
   MOZ_ALWAYS_SUCCEEDS(NS_DispatchToMainThread(runnable));
